tests/ui/try-trait/try-after-unwrap-issue-127345.rs

```diff
@@ -0,0 +1,39 @@
+fn foo() -> Option<usize> {
+    let x = Some(42).expect("moop")?;
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.expect()`
+    let x = Some(42).unwrap()?;
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.unwrap()`
+    x
+}
+
+fn bar() -> Option<usize> {
+    foo().or(Some(43)).unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `usize`
+    //~| HELP remove the `.unwrap()`
+}
+
+fn baz() -> Result<usize, ()> {
+    Ok(44).unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `{integer}`
+    //~| HELP remove the `.unwrap()`
+}
+
+struct FakeUnwrappable;
+
+impl FakeUnwrappable {
+    fn unwrap(self) -> () {}
+}
+
+fn qux() -> Option<usize> {
+    FakeUnwrappable.unwrap()?
+    //~^ ERROR the `?` operator can only be applied to values that implement `Try`
+    //~| HELP the trait `Try` is not implemented for `()`
+}
```

Comment:
Please add a test for implementing an extension trait for `Option` that also uses `unwrap()`. Also add tests where the predicate does not hold when unwrap is removed.

Reply:
The extension trait test is unnecessary since resolution always prefers inherent methods first.

Reply:
I'm assuming if you are using ? in a function that doesn't support Try or supports try for a different type it could fail. Even if that is not how it works please add that as a test

Reply:
Done, thanks!